core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOInfo.scala

```diff
@@ -0,0 +1,80 @@
+package org.bitcoins.core.wallet.utxo
+
+import org.bitcoins.core.currency.CurrencyUnit
+import org.bitcoins.core.protocol.transaction.{
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.crypto.Sign
+
+sealed trait UTXOInfo[+InputType <: InputInfo] {
+  def inputInfo: InputType
+  def hashType: HashType
+  def signers: Vector[Sign]
+
+  private val keysToSignFor = inputInfo.pubKeys
+  require(signers.map(_.publicKey).forall(keysToSignFor.contains),
+          s"Cannot have signers that do not sign for one of $keysToSignFor")
+
+  def amount: CurrencyUnit = inputInfo.amount
+  def output: TransactionOutput = inputInfo.output
+  def outPoint: TransactionOutPoint = inputInfo.outPoint
+  def conditionalPath: ConditionalPath = inputInfo.conditionalPath
+}
+
+object UTXOInfo {
+  type Any = UTXOInfo[InputInfo]
```

Comment:
nit: I would prefer not to use type aliases unless the names are really long. These seem reasonable to me